walk-in-interview/src/main/webapp/new-job.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * This file is specific to new-job.html. It renders the fields on the
+ * page dynamically, and it also makes the POST request when the form
+ * is submitted.
+ */
+
+const NEW_JOB_TITLE_ID = 'new-job-title';
+const NEW_JOB_ADDRESS_ID = 'new-job-address';
+const NEW_JOB_DESCRIPTION_ID = 'new-job-description';
+const NEW_JOB_PAY = {
+  FREQUENCY_ID: 'new-job-pay-frequency',
+  MIN_ID: 'new-job-pay-min',
+  MAX_ID: 'new-job-pay-max',
+};
+const REQUIREMENTS_LIST_NAME = 'requirements-list';
+const NEW_JOB_EXPIRY_ID = 'new-job-expiry';
+const NEW_JOB_DURATION_ID = 'new-job-duration';
+
+const HOMEPAGE_PATH = '/walk-in-interview/src/main/webapp/index.html';
+
+/**
+ * Adds a new job listing by making a POST request to the /jobs servlet.
+ * @return {Promise} Fetch function or promise with reject if invalid params.
+ */
+function addJob() {
+  const name = document.getElementById(NEW_JOB_TITLE_ID).value;
+  const address = document.getElementById(NEW_JOB_ADDRESS_ID).value;
+  const description = document.getElementById(NEW_JOB_DESCRIPTION_ID).value;
+  const payFrequency = document.getElementById(NEW_JOB_PAY.FREQUENCY_ID).value;
+
+  const payMin = document.getElementById(NEW_JOB_PAY.MIN_ID).valueAsNumber;
+  const payMax = document.getElementById(NEW_JOB_PAY.MAX_ID).valueAsNumber;
+  if (payMin > payMax) {
+    return Promise.reject(new Error('min greater than max'));
```

Comment:
`Promise` is used for async stuff like a network call example `fetch` method below. I'd recommend separating non async tasks as the two methods with different responsibilities

```js
/** Gets job detail from user input. */
function getJobDetailsFromUserInput() {}

/** Function validate input */
function validateUserInput() {}
```

Now consume it like
```js
const submitButton = document.getElementById('new-job-submit');
submitButton. addEventListener('click', (_) => {
  if (!validateUserInput()) {
    // show some error UI
    return;
  }
 
  const jobDetails = getJobDetailsFromUserInput();
  fetch('/jobs', {
    method: 'POST',
    headers: {'Content-Type': 'application/json'},
    body: JSON.stringify(jobDetails),
   })
   .then((repsonse) => repsonse.text())
   .then((data) => {
      console.log('data', data);
      window.location.href= HOMEPAGE_PATH;
    })
    .catch((error) => {
      // some error ui
      console.log('error', error);
    });
});


```